resources/template/pageTemplate.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace template;
+
+require $_SERVER['DOCUMENT_ROOT'] . '/../vendor/autoload.php';
+
+use src\ProjectConfig;
+use src\SessionCookie;
+use template\Menu;
+
+class Page
+{
+    public $config;
+    public $session;
+    public $head;
+    public $menu;
+    public $content;
+    public $foot;
+    public function __construct()
+    {
+        $this->config = new ProjectConfig;
+        $this->session = new SessionCookie;
+        $title = $this->config->pageDict[basename($_SERVER['PHP_SELF'], '.php')];
+
+        $this->head = "
+        <!DOCTYPE html>
+        <html lang='en'>
+        <meta name='Description' content='Bookhive - A Digital Library Management System'>
+        <meta name='viewport' content='width=device-width, initial-scale=1'>
+        <title>" . $title . "</title>
+        <link rel='stylesheet' href='https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css'>
+        <link rel='stylesheet' href='" . $this->config->config["paths"]["css"] . "/main.css'>
+        </head>
+        <body>
+        ";
+
+        if (isset($_SESSION["logged_in"])) {
+            $this->menu = new Menu(basename(__FILE__), $_SESSION["admin"], $_SESSION["uname"]);
+        } else {
+            $this->menu = new Menu(basename(__FILE__));
+        }
```

Comment:
Use session class here. Also verify once that `basename(__FILE__)` logic is working correctly

Reply:
Mmm okay just a min

Reply:
@shriaas2898 I have broken my code just now!
Its your logic na, can you verify if its working?

Reply:
fine the menu is broken and the highlighter is not working as expected

Reply:
@AJV009  I have tested the `basename(__FILE__)` it works fine can you explain where exactly the problem is?

Reply:
@shriaas2898  thanks, but @guptahemant  doubted if it worked! probably he knows some kind of best practice or some suggestion, do you?
SEE as far as I know the linking is working its proper and has no issues BUT the menu is not collapsing and not highlighting as expected.

Reply:
I can fix the collapsing part I'll look to find out why active menu is not getting highlighted

Reply:
@AJV009  I have fixed the collapsible menu in #106

Reply:
@guptahemant  can you take a look!

Reply:
Not using session class here?

Reply:
@guptahemant  do you think its needed its just one line -> `if (isset($_SESSION["logged_in"])) {`
Wait did you mean put the whole block in session class and call em here?
```php
        if (isset($_SESSION["logged_in"])) {
            $this->menu = new Menu(basename(__FILE__), $_SESSION["admin"], $_SESSION["uname"]);
        } else {
            $this->menu = new Menu(basename(__FILE__));
        }
```

Reply:
i meant instead of using $_SESSION variable here, you should utilize session class to check if user is logged in or not, In that way if you need to change the logic for login then you need to change the logic inside that method only and not the whole codebase.

Reply:
mmm okay!

Reply:
@guptahemant  am still bit confused on what you actually mean!
DO you that I should something like ->
```
if ....
return true
else false
```
OR add this block to session class
```php
if (isset($_SESSION["logged_in"])) {
            $this->menu = new Menu(basename(__FILE__), $_SESSION["admin"], $_SESSION["uname"]);
        } else {
            $this->menu = new Menu(basename(__FILE__));
        }
```

Reply:
In session class:
Method: is_user_logged_in() -> Returns true / False.

Then the updated logic will be like:
$session_obj = new Session;
if($session_obj->is_user_logged_in()) {
// some action
}
else {
// Some action
}

Reply:
basically the option 1 you mentioned.

Reply:
fine thats what I asked first


> @guptahemant do you think its needed its just one line -> `if (isset($_SESSION["logged_in"])) {`
> Wait did you mean put the whole block in session class and call em here?
> 
> ```
>         if (isset($_SESSION["logged_in"])) {
>             $this->menu = new Menu(basename(__FILE__), $_SESSION["admin"], $_SESSION["uname"]);
>         } else {
>             $this->menu = new Menu(basename(__FILE__));
>         }
> ```

:laughing:  :laughing:  :laughing: 
I understood what you meant now!